ccdproc/core.py

```diff
@@ -703,6 +703,7 @@ def gain_correct(ccd, gain, gain_unit=None):
         gain_value = gain
 
     result = ccd.multiply(gain_value)
+    result.meta = ccd.meta.copy()
```

Comment:
Now that you mention it, I investigated a bit and found why this change was necessary:

The `_ccddata_arithmetic` makes sure the meta is kept when the other operand is a scalar (not CCDData), the `NDArithmeticMixin` however defaults (because of backwards-compatibility-reasons) to no discarding the `meta`. 
Note that currently operations between two CCDData independant of the astropy version discard their meta, it was just scalars that had the special case.